src/core/net/dns_types.cpp

```diff
@@ -221,20 +222,38 @@ Error Name::AppendTerminator(Message &aMessage)
 
 Error Name::AppendPointerLabel(uint16_t aOffset, Message &aMessage)
 {
+    Error    error;
+    uint16_t value;
+
+#if OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE
+    if (!Instance::IsDnsNameCompressionEnabled())
```

Comment:
Should the `IsDnsNameCompressionEnabled` be defined in the `Dns::Name` class? And the method name can be shorter I think :)

Reply:
I was thinking of the same (I agree the method name would be nicer/shorter), but I thought that it would be better to keep the `static` member definitions in `Instance` so we can more easily track them. 

My view is that we should avoid `static` member definitions in core module unless there is a good reason and if we use them we should aim to keep them in one place/module (`Instance` makes sense) to make easy to track them. Otherwise the `static` member definition can be added/hidden is some some type and harder to track.

Thoughts?

Reply:
Thanks, makes sense to me!